Include/cpython/compile.h

```diff
@@ -0,0 +1,99 @@
+#ifndef Py_CPYTHON_COMPILE_H
+#define Py_CPYTHON_COMPILE_H
+
+#ifndef Py_LIMITED_API
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
I suggest to remove these guards, and instead use something like:

```
#ifndef Py_CPYTHON_ABSTRACTOBJECT_H
#  error "this header file must not be included directly"
#endif
```

As one by other header files.

Reply:
done.